code/game/objects/items/storage/belt.dm

```diff
@@ -266,6 +266,100 @@
 	var/datum/component/storage/STR = GetComponent(/datum/component/storage)
 	STR.max_items = 6
 
+/obj/item/storage/belt/security/nemesis
+	name = "power belt"
+	desc = "A tactical black belt with a blue strip and a battery, designed to be used along with stun gloves"
+	icon_state = "nemesis"
+	worn_icon_state = "nemesis"
+	var/overcharge = 0
+	//Timer for charge depletion
+	var/charge_depletion = 0
+	//Overlay that is added to the owner if overcharged
+	var/mutable_appearance/mob_overlay
+
+
+/obj/item/storage/belt/security/nemesis/ComponentInitialize()
+	. = ..()
+	var/datum/component/storage/STR = GetComponent(/datum/component/storage)
+	STR.set_holdable(list(
+		/obj/item/melee/baton,
+		/obj/item/melee/classic_baton,
+		/obj/item/grenade,
+		/obj/item/reagent_containers/spray/pepper,
+		/obj/item/restraints/handcuffs,
+		/obj/item/assembly/flash/handheld,
+		/obj/item/clothing/glasses,
+		/obj/item/ammo_casing/shotgun,
+		/obj/item/ammo_box,
+		/obj/item/food/donut,
+		/obj/item/kitchen/knife/combat,
+		/obj/item/flashlight/seclite,
+		/obj/item/melee/classic_baton/telescopic,
+		/obj/item/radio,
+		/obj/item/clothing/gloves,
+		/obj/item/restraints/legcuffs/bola,
+		/obj/item/holosign_creator/security,
+		/obj/item/nemesis_mine,
+		/obj/item/nemesis_trap,
+		))
+
+/obj/item/storage/belt/security/nemesis/full/PopulateContents()
+	new /obj/item/reagent_containers/spray/pepper(src)
+	new /obj/item/restraints/handcuffs(src)
+	new /obj/item/restraints/handcuffs(src) //Stunbatons are useless so we give them additional handcuffs
+	new /obj/item/grenade/flashbang(src)
+	new /obj/item/assembly/flash/handheld(src)
+
+/obj/item/storage/belt/security/nemesis/Initialize()
+	. = ..()
+	mob_overlay = mutable_appearance('icons/effects/effects.dmi', "overcharged", BELOW_MOB_LAYER)
+
+/obj/item/storage/belt/security/nemesis/proc/update_charge(charge = 0)
+	var/charge_icon = 0
+	if((charge + overcharge) / NEMESIS_MAX_CHARGE >= 1)
+		charge_icon = 100
+	else if((charge + overcharge) / NEMESIS_MAX_CHARGE >= 0.75)
+		charge_icon = 75
+	else if((charge + overcharge) / NEMESIS_MAX_CHARGE >= 0.5)
+		charge_icon = 50
+	else if((charge + overcharge) / NEMESIS_MAX_CHARGE >= 0.25)
+		charge_icon = 25
+	icon_state = "nemesis[charge_icon]"
+	worn_icon_state = "nemesis[charge_icon]"
+	update_icon()
+
+	if(overcharge)
+		if(ishuman(loc))
+			var/mob/living/carbon/human/owner = loc
+			owner.overlays += mob_overlay
+			owner.update_appearance()
+	else
+		if(ishuman(loc))
+			var/mob/living/carbon/human/owner = loc
+			owner.overlays -= mob_overlay
+			owner.update_appearance()
+
+	if(charge + overcharge > 0)
+		START_PROCESSING(SSobj, src)
+
+/obj/item/storage/belt/security/nemesis/process(delta_time)
+	charge_depletion++
+	if(charge_depletion >= 5 SECONDS)
```

Comment:
uhhhhhhhhhhhhhhh

5 SECONDS = 50, so process() would have to run 50 times (NOT factoring in delta_time) for the code in this if statement to be run